tests/utils/tests.py

```diff
@@ -141,3 +142,8 @@ def x(self, x):
 
 def test_get_name_from_func_lambda():
     assert "tests.utils.tests.<lambda>" == get_name_from_func(lambda x: "x")
+
+
+def test_url_sanitization():
+    sanitized = sanitize_url("http://user:pass@localhost:123/foo?bar=baz#bazzinga")
```

Comment:
Would you mind throwing in some percent-encoded bytes in the password?

It appears that urlparse doesn't unescape them, but it would be good to have a test to ensure that remains the case over time, in case the implementation of sanitize_url changes.

Reply:
added in 388d6c2c